src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,286 @@
+    SynapseClient.detectSSOCode()
+    return SynapseClient.getSessionTokenFromCookie()
+      .then((sessionToken: any) => { return this.refresh(sessionToken) })
+      .catch((error: any) => {
+        console.error(error)
+      })
+  }
+
+
+  refresh = (token?: string): Promise<any> => {
+    this.setState({ token })
+    if (token) {
+      return this.getUserFileListing(token!, this.props.parentContainerId).catch((error) => {
+        this.onError(error)
+      })
+
+    } else {
+      return Promise.reject()
+    }
+  }
+
+
+
+  private getUserFileListing = async (token: string, parentContainerId: string): Promise<UserData> => {
+    this.setState({ isLoading: true });
+    const result: UserData = {
+      fileList: [],
+      existingData: {},
+      dataFolderId: ''
+    }
+
+    // get user profile
+    const userprofile = await SynapseClient.getUserProfile(token)
+
+    //find directory named with profileId
+    const entityLookupRequest: EntityLookupRequest = {
+      entityName: userprofile.ownerId,
+      parentId: parentContainerId
+    }
+    try {
+      const entity = await SynapseClient.lookupChildEntity(entityLookupRequest, token)
+
+      //if found - that's the folder id
+      result.dataFolderId = entity.id
+    }
+
+    catch (error) {
+      if (error.statusCode === 404) {
+        // if doesn't exist - create
+        const entityId = await this.createDataFolder(userprofile, token)
+        if (!entityId) {
+          throw ('no folder')
+        }
+        result.dataFolderId = entityId;
+      } else {
+        this.onError(error)
+        return Promise.reject(error);
+      }
+    }
+    finally {
+      // get the file listing
+      result.fileList = await this.getFileListingInFolder(token, result.dataFolderId)
+      this.setState({
+        userprofile,
+        dataFolderId: result.dataFolderId,
+        fileList: result.fileList || [],
+        isLoading: false
+      })
+      return result;
+    }
+  }
+  onError = (args: any) => {
+    console.log(args)
+  }
+
+
+  // creates data folder for user
+  createDataFolder = async (userprofile: UserProfile, token: string): Promise<string | undefined> => {
+
+    const newFolder: Entity = {
+      parentId: this.props.parentContainerId,
+      name: userprofile.ownerId,
+      concreteType: 'org.sagebionetworks.repo.model.Folder',
+    }
+    try {
+      const folder: Entity = await SynapseClient.createEntity(newFolder, token);
+      return folder.id
+    }
+    catch (error) {
+      this.onError(error)
+      throw (error)
+    }
+  }
+
+
+
+  getFileListingInFolder = async (token: string, targetFolderId: string): Promise<Entity[]> => {
+    const request = {
+      includeTypes: ['file'],
+      parentId: targetFolderId,
+      sortBy: 'NAME',
+      sortDirection: 'ASC'
+    }
+
+    try {
+      const data = await SynapseClient.getEntityChildren(request, token);
+      return data.page;
```

Comment:
Looks like this is only getting the first page of results.  Maybe this is ok for initial release, but we should confirm (and likely add a "Show more" button to get the next page of data).  Are previously submitted items shown?  If not, then there are a few possible solutions.  You create a subfolder (Submissions) and move the entity into there after successful submission (then they would not be returned by your getChildren query).  Or we could utilize annotations.

Reply:
You are correct. This is still outstanding functionality.  Will try to get it into v1. This is somewhat of a WIP. Do you have a preference as to subfolder vs annotation?